services/audit-service/src/main/java/datawave/microservice/audit/auditors/file/config/FileAuditConfig.java

```diff
@@ -0,0 +1,51 @@
+package datawave.microservice.audit.auditors.file.config;
+
+import datawave.microservice.audit.auditors.file.FileAuditor;
+import datawave.microservice.audit.config.AuditProperties;
+import datawave.webservice.common.audit.Auditor;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+import javax.validation.Valid;
+import java.util.List;
+
+/**
+ * Configures an FileAuditor to process messages received by the audit service in the case that our messaging infrastructure has failed. This configuration is
+ * activated via the 'audit.auditors.file.enabled' property.
+ *
+ */
+@Configuration
+@ConditionalOnProperty(name = "audit.auditors.file.enabled", havingValue = "true")
```

Comment:
how is this different than the dump auditor?

Reply:
The file auditor our last ditch attempt at auditing, used when all other attempts to audit have failed (at the audit rest endpoint).  

The dump auditor is listening to the audit.dump queue, and writes any messages moved there to disk.  So, if you have some stubborn messages in your DLQ that you need to analyze, you can move them to the audit.dump queue, and then work with the file on disk.